src/components/ToDoItem.tsx

```diff
@@ -19,33 +19,41 @@ export const ToDoItem: React.FC<ToDoItem> = ({
   newTextWithId,
 }) => {
   const isEdit = newTextWithId?.id === item.id;
+  
+  const onInputChangeNewTextWithId = React.useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) =>
+      setNewTextWithId({
+        id: newTextWithId?.id ?? 0,
```

Comment:
Please, try using `previousState` instead of passing it directly. I'm not certain whether it would work, but we should double check.